pyfr/integrators/dual/pseudo/multip.py

```diff
@@ -241,15 +240,10 @@ def restrict(self, l1, l2):
         # mg1 = Q^ns
         self.pintg._add(0, mg1, 1, l2idxcurr)
 
-        # Restrict the physical stepper terms
-        for i in range(self.pintg.stepper_nregs):
-            l1sys.eles_scal_upts_inb.active = (
-                self.pintgs[l1]._stepper_regidx[i]
-            )
-            l2sys.eles_scal_upts_inb.active = (
-                self.pintgs[l2]._stepper_regidx[i]
-            )
-            self.pintg._queue.enqueue_and_run(self.mgproject(l1, l2))
+        # Restrict the physical source term
```

Comment:
So now the lower levels do not require storage for the old and previous states, only the extra register is in use. Can we easily remove these unused registers to save memory?

Reply:
We should be able to, and it is probably worth doing.